plugins/flytekit-auto-cache/README.md

```diff
@@ -0,0 +1,58 @@
+# Flyte Auto Cache Plugin
+
+This plugin provides a caching mechanism for Flyte tasks that generates a version hash based on the source code of the task and its dependencies. It allows users to manage the cache behavior.
+
+## Usage
+
+To install the plugin, run the following command:
+
+```bash
+pip install flytekitplugins-auto-cache
+```
+
+To use the caching mechanism in a Flyte task, you can define a `CachePolicy` that combines multiple caching strategies. Here’s an example of how to set it up:
+
+```python
+from flytekit import task
+from flytekit.core.auto_cache import CachePolicy
+from flytekitplugins.auto_cache import CacheFunctionBody, CachePrivateModules
+
+cache_policy = CachePolicy(
+    auto_cache_policies = [
+        CacheFunctionBody(),
+        CachePrivateModules(root_dir="../my_package"),
+        ...,
+    ]
+    salt="my_salt"
+)
```

Comment:
let's also provide an example of not needing to provide a `CachePolicy` object, e.g. just a passing in `CacheFunctionBody`.